### DISCUSSION TITLE
Way to extend an expiring session token

### DISCUSSION THREAD

**Discussion by ramfree17** [gravitational/teleport#10234] [Q&A] [answered]
Is there a way from the command line to request the authenticating server to extend or replace my current session's token? This is assuming that there are a few minutes more remaining in my current session. 

Currently I just re-login and go through the multi-factor process, but given my remote location the login process is noticeable slow. Extending the session timeout is beyond my control so I am hoping I can minimize the disruption in my flow if I can already proactively renew/replace the session token that is about to expire. 

Thanks.

**Comment by webvictim** [comment, marked as answer]
There isn't currently any way to do this with Teleport - you will need to reauthenticate completely once your session has expired. You can always run `tsh login` again at any point to get a new certificate with an updated TTL, which basically accomplishes the same goal.

The reason is that Teleport needs to open a web browser to handle reauthentication with an SSO provider (in case your user there has been deactivated etc) and the experience of opening a web browser to extend the token is more or less exactly the same as logging in from scratch again anyway.

**Comment by mkpankov** [comment]
I'd like to add to this. Sometimes I work in `ssh` w/o any terminal multiplexer like `tmux` and getting disconnected b/c of expiring token is annoying. It can happen in middle of work day, because I can start the session in the middle of previous day. Rather, I'd like to extend my session to cover current day, in the morning.
